src/main/scala/cromwell/engine/CallExecutionActor.scala

```diff
@@ -64,22 +63,33 @@ class CallExecutionActor(backendCall: BackendCall) extends Actor with CromwellAc
     .setMultiplier(1.1)
     .build()
 
-  /** Intended for use with `Future#onComplete`, if the `Future` completes successfully apply `successFunction`
-    * to the result value.  If the `Future` is failed, log and message self to `Finish`. */
-  def ifSuccess[T](successFunction: T => Unit): Try[T] => Unit = {
-    case Success(s) => successFunction(s)
-    case Failure(t) =>
-      logger.error(t.getMessage, t)
-      self ! Finish(FailedExecutionHandle(t))
+  /**
+   * If the `work` `Future` completes successfully, perform the `onSuccess` work, otherwise schedule
+   * the execution of the `onFailure` work using an exponential backoff.
+   */
+  def withRetry(work: Future[ExecutionHandle], onSuccess: ExecutionHandle => Unit, onFailure: => Unit): Unit = {
+    work onComplete {
+      case Success(s) => onSuccess(s)
+      case Failure(e: Exception) =>
+        logger.error(e.getMessage, e)
+        scheduleWork(onFailure)
+      case Failure(throwable) =>
+        logger.error(throwable.getMessage, throwable)
+        throw throwable
```

Comment:
Are you catching throwable here specifically for Errors? i.e. JVM level badness?